src/components/Modals/AddContactModal.jsx

```diff
@@ -104,6 +143,15 @@ const AddContactModal = ({ addContact, showAddContactModal, setShowAddContactMod
     }
 
     try {
+      if (userObject.webId !== originalWebId && contactToEdit) {
+        if (await contactWebIds.includes(userObject.webId)) {
+          addNotification('error', 'Web ID exists. Edit appropriate contact');
+          return;
+        }
+        const toDelete = contacts.find((item) => item.webId === originalWebId);
+        await handleDeleteContact(toDelete);
+        setDeleteViaEdit(true); // attempt to re-render
```

Comment:
This function seems to delete the contact when editing and submitting without a username by mistake.

https://github.com/codeforpdx/PASS/assets/14917816/dd9e99de-a1b1-4644-af9e-3787a8f88cef

Seems to be related to a validation bug we found with the modal (see Issue #614), but we probably don't want the contact to be deleted if the information is incorrect, just a warning and error stating invalid information.